### PR TITLE
Doxygen improvements

### PR DESCRIPTION
**Pull request by evincarofautumn**
This is a large but non-functional change. I turned off Doxygen’s `EXTRACT_ALL` mode, added explicit `\file` commands to the files we want documented, and fixed a couple of errors. This gives us proper warnings about undocumented members.

I chose to remove the filenames (i.e., not use `\file filename.c`) because it’s redundant and tends to get out of date when files are moved—some of them were already outdated or misspelled.

### CODE REVIEW

**Comment by kumpera** [comment]
How do we tell apart what needs to be documented from what doesn't?

File: mono/dis/util.c [hunk 1]
```diff
@@ -20,9 +20,8 @@
 #endif
 
 /**
- * map:
- * @code: code to lookup in table
- * @table: table to decode code
+ * \param code code to lookup in table
```
Comment (kumpera) [inline mono/dis/util.c]: Does this works with docs.mono-project.com?

Comment (evincarofautumn) [inline mono/dis/util.c]: I don’t know. I thought we weren’t generating anything from the runtime sources. Also, that URL doesn’t resolve; are you referring to http://www.mono-project.com/docs/ or something else?

Comment (kumpera) [inline mono/dis/util.c]: Did we lose the existing page with embedding API docs?

Comment (evincarofautumn) [inline mono/dis/util.c]: Ah, there is http://docs.go-mono.com/?link=root:/embed linked from http://www.mono-project.com/docs/advanced/embedding/ but I’m not sure how it’s generated.

Comment (kumpera) [inline mono/dis/util.c]: It's ok, ask around.
We can't lose a way to publish the existing docs on the embedding API alone (no internals).

**Comment by evincarofautumn** [comment]
> How do we tell apart what needs to be documented from what doesn't?

I plan to (write some Perl to) convert existing docs to Doxygen syntax, then see how many warnings remain and whether they should be fixed or just disabled.

**Comment by vargaz** [comment]
Putting the filename in the initial comment is a convention for a lot of projects. No idea why tough.

**Comment by evincarofautumn** [comment]
It might be apocryphal, but I once heard it was so that you could find the original file, given a source listing printed on paper.

**Comment by evincarofautumn** [comment]
I modified `exdoc` to support a subset of Doxygen syntax, semi-automatically updated all the API docs (except for `mono-api-wapi`), and verified that we get reasonable results from both doc generators. I found a bunch of seemingly dead APIs that I’ll verify and remove in a separate patch.

**Pull request merged**